tests/dialects/test_singlestore.py

```diff
@@ -73,3 +73,89 @@ def test_cast(self):
         self.validate_identity("SELECT '{\"a\" : 1}' :> JSON")
         self.validate_identity("SELECT NOW() !:> TIMESTAMP(6)")
         self.validate_identity("SELECT x :> GEOGRAPHYPOINT")
+
+    def test_json_extract(self):
+        self.validate_identity(
+            "SELECT a::b FROM t", write_sql="SELECT JSON_EXTRACT_JSON(a, 'b') FROM t"
+        )
+        self.validate_identity(
+            "SELECT a::$b FROM t", write_sql="SELECT JSON_EXTRACT_STRING(a, 'b') FROM t"
+        )
+        self.validate_identity(
+            "SELECT a::%b FROM t", write_sql="SELECT JSON_EXTRACT_DOUBLE(a, 'b') FROM t"
+        )
+        self.validate_identity(
+            "SELECT a::`b`::`2` FROM t",
+            write_sql="SELECT JSON_EXTRACT_JSON(JSON_EXTRACT_JSON(a, 'b'), 2) FROM t",
+        )
+        self.validate_identity(
+            "SELECT a::2 FROM t", write_sql="SELECT JSON_EXTRACT_JSON(a, 2) FROM t"
+        )
+
+        self.validate_all(
+            "SELECT JSON_EXTRACT_JSON(a, 'b') FROM t",
+            read={"": "SELECT JSON_EXTRACT(a, '$.b') FROM t"},
+            write={"": "SELECT JSON_EXTRACT(a, '$.b') FROM t"},
```

Comment:
Why are we testing with `""` instead of a concrete dialect? Let's ensure roundtrip tests are always included by having `"singlestore"` as either a read or write target.